src/server/acl/user_registry.cc

```diff
@@ -86,11 +87,11 @@ void UserRegistry::Init() {
   auto default_user = DefaultUserUpdateRequest();
   auto maybe_password = absl::GetFlag(FLAGS_requirepass);
   if (!maybe_password.empty()) {
-    default_user.password = std::move(maybe_password);
+    default_user.passwords.front().nopass = false;
```

Comment:
should not you set `maybe_password` as well?

Reply:
this is a bug! it also got caught by the tests ;)